mobile/packages/swap/adapters/api/minswap/transformers.ts

```diff
@@ -0,0 +1,364 @@
+      return Dex.Spectrum
+    case Swap.Protocol.Splash_v1:
+      return Dex.SplashStable
+    default:
+      return Dex.Unsupported
+  }
+}
+
+export const transformersMaker = (config: MinswapApiConfig) => {
+  const {isPrimaryToken, primaryTokenInfo, address, partner} = config
+
+  // Convert portfolio token ID to API token ID (like DexHunter)
+  const toTokenId = (tokenId: Portfolio.Token.Id) => {
+    const result = isPrimaryToken(tokenId)
+      ? 'lovelace'
+      : tokenId.replace('.', '')
+    return result
+  }
+
+  const transformToken = (token: {
+    token_id: string
+    logo: string | null
+    ticker: string | null
+    is_verified: boolean | null
+    price_by_ada: number | null
+    project_name: string | null
+    decimals: number | null
+  }): Portfolio.Token.Info => {
+    const formattedTokenId: Portfolio.Token.Id =
+      token.token_id === 'lovelace'
+        ? primaryTokenInfo.id
+        : `${token.token_id.slice(0, 56)}.${token.token_id.slice(56)}`
+
+    return freeze(
+      {
+        id: formattedTokenId,
+        name: token.project_name || token.ticker || 'Unknown Token',
+        ticker: token.ticker || '',
+        decimals: token.decimals || 6,
+        logo: token.logo || null,
+        description: '',
+        website: '',
+        policyId:
+          token.token_id === 'lovelace' ? '' : token.token_id.slice(0, 56),
+        fingerprint: '',
+        group: token.token_id === 'lovelace' ? 'ADA' : null,
+        kind: token.token_id === 'lovelace' ? 'ft' : 'ft',
+        image: token.logo || null,
+        icon: token.logo || null,
+        symbol: token.ticker || '',
+        metadatas: {},
+        isPrimaryToken: isPrimaryToken(formattedTokenId),
+        status: Portfolio.Token.Status.Valid,
+        application: Portfolio.Token.Application.General,
+        tag: '',
+        reference: '',
+        originalImage: '',
+        nature: Portfolio.Token.Nature.Secondary,
+        type: Portfolio.Token.Type.FT,
+      },
+      true,
+    )
+  }
+
+  // Transform Minswap paths to Swap.Split format
+  const transformPathsToSplits = (paths: Array<any>): Swap.Split[] => {
+    if (!paths || paths.length === 0) return []
+
+    return paths
+      .map((path) => {
+        // Each path is an array of hops
+        const firstHop = path[0]
+        if (!firstHop) return null
+
+        const split = freeze(
+          {
+            amountIn: parseFloat(firstHop.amount_in || '0'),
+            batcherFee: 0, // Minswap doesn't provide batcher fee per split
+            deposits: parseFloat(firstHop.deposits || '0'),
+            protocol: mapDexToProtocol(firstHop.protocol),
+            expectedOutput: parseFloat(firstHop.amount_out || '0'),
+            expectedOutputWithoutSlippage: parseFloat(
+              firstHop.min_amount_out || '0',
+            ),
+            fee: parseFloat(firstHop.dex_fee || '0'),
+            initialPrice:
+              parseFloat(firstHop.amount_out || '0') /
```

Comment:
can parseFloat handle BN? I think it should use the helper from common

Reply:
Those are not BN. But I'm removing the parseFloat anyway.